kore/src/Kore/Strategies/Goal.hs

```diff
@@ -1077,3 +1096,33 @@ configurationDestinationToRule ruleType configuration rhs =
         , rhs
         , attributes = Default.def
         }
+
+debugProofStateBracket
+    :: MonadLog log
+    => ToRulePattern goal
+    => ToRulePattern (Rule goal)
+    => ProofState goal goal ~ ProofState.ProofState goal
+    => Prim goal ~ ProofState.Prim (Rule goal)
+    => ProofState goal goal
+    -- ^ current configuration
+    -> Prim goal
+    -- ^ transition
+    -> log (ProofState goal goal)
+    -- ^ action to be computed
+    -> log (ProofState goal goal)
+debugProofStateBracket
+    (fmap toRulePattern -> config)
+    (fmap toRulePattern -> transition)
+    action
+  = do
+    logM DebugProofState
+        { configuration = Before config
+        , transition
+        }
+    result <- action
+    logM DebugProofState
```

Comment:
We never log an `After` without a `Before`, right? I think the log would be more useful if we logged the initial and final proof states together. Otherwise, I don't think there is an easy way to associate them with each other. Could we change the entry type to be `DebugProofState { initialProofState, finalProofState, transition }`?